### PR TITLE
 Add support for generic webhooks 

### PR DESCRIPTION
**Pull request by nijel**
This can be useful with push notification services such as Gotify.

Built on top of #95

### CODE REVIEW

**Comment by mtakaki** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ cachet_url_monitor/configuration.py  2
         

Coverage decreased per file
===========================
- cachet_url_monitor/scheduler.py  -2
         

Complexity increasing per file
==============================
- cachet_url_monitor/webhook.py  2
- cachet_url_monitor/configuration.py  1
         

Complexity decreasing per file
==============================
+ tests/test_configuration.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mitsuotakaki/cachet-url-monitor/pullRequest?prid=5439800&bid=17838996)

File: cachet_url_monitor/configuration.py [hunk 1]
```diff
@@ -45,12 +47,14 @@ class Configuration(object):
...
     message: str
 
-    def __init__(self, config, endpoint_index: int, client: CachetClient, token: str):
+    def __init__(self, config, endpoint_index: int, client: CachetClient, webhooks : Optional[List[Webhook]] = None):
         self.endpoint_index = endpoint_index
         self.data = config
         self.endpoint = self.data['endpoints'][endpoint_index]
         self.client = client
-        self.token = token
+        self.webhooks = []
```
Comment (mtakaki) [inline cachet_url_monitor/configuration.py:55]: You could actually do:
```
self.webhooks = webhooks or []
```

File: cachet_url_monitor/configuration.py [hunk 2]
```diff
@@ -239,6 +239,25 @@ def push_metrics(self):
...
+        if self.status == st.ComponentStatus.PERFORMANCE_ISSUES:
+            message = self.message
+            title = f'{self.endpoint["name"]} degraded'
+        elif self.status == st.ComponentStatus.OPERATIONAL:
+            message = 'Incident resolved'
+            title = f'{self.endpoint["name"]} OK'
+        else:
+            message = self.message
+            title = f'{self.endpoint["name"]} unavailable'
+        print('hooks', self.webhooks)
```
Comment (mtakaki) [inline cachet_url_monitor/configuration.py:253]: We should use logger here:
```
self.logger.info(f'hooks {self.webhooks}')
```

File: cachet_url_monitor/webhook.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+class Webhook:
+    url: str
+    params: Dict[str, str]
+
+    def __init__(self, url: str, params: Dict[str, str]):
+        self.url = url
+        self.params = params
+
+    def push_incident(self, title: str, message: str):
+        format_args = {
```
Comment (mtakaki) [inline cachet_url_monitor/webhook.py:15]: What do you think about also sharing the component id?

File: cachet_url_monitor/configuration.py (hunk 2 above)
Comment (nijel) [reply cachet_url_monitor/configuration.py:253]: Better remove this debugging code, this is not really useful thing to log.

File: cachet_url_monitor/webhook.py (hunk 3 above)
Comment (nijel) [reply cachet_url_monitor/webhook.py:15]: I didn't find any use for it here. There are many variables which might be useful to include for the interpolation, but I'd do that once somebody needs them.

**Comment by mtakaki** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage increased per file
===========================
+ cachet_url_monitor/configuration.py  3
         

Coverage decreased per file
===========================
- cachet_url_monitor/scheduler.py  -2
         

Complexity increasing per file
==============================
- cachet_url_monitor/configuration.py  1
- cachet_url_monitor/webhook.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mitsuotakaki/cachet-url-monitor/pullRequest?prid=5439800&bid=17838996)

File: cachet_url_monitor/webhook.py (hunk 3 above)
Comment (mtakaki) [reply cachet_url_monitor/webhook.py:15]: Sounds good to me.

**Comment by mtakaki** [comment]
Thank you once again for this awesome contribution :)

**Pull request merged**